subprojects/logging/src/integTest/groovy/org/gradle/internal/logging/BasicGroupedTaskLoggingFunctionalSpec.groovy

```diff
@@ -0,0 +1,115 @@
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.gradle.internal.logging
+
+import org.gradle.integtests.fixtures.AbstractConsoleFunctionalSpec
+import spock.lang.Issue
+
+class BasicGroupedTaskLoggingFunctionalSpec extends AbstractConsoleFunctionalSpec {
+
+    def "multi-project build tasks logs are grouped"() {
+        given:
+        settingsFile << "include '1', '2', '3'"
+
+        buildFile << """
+            subprojects {
+                task log { doFirst { logger.quiet "Output from " + project.name } }
+            }
+        """
+
+        when:
+        succeeds('log')
+
+        then:
+        groupedOutputs.size() == 3
+        groupedOutputs[':1:log'] == "Output from 1"
+        groupedOutputs[':2:log'] == "Output from 2"
+        groupedOutputs[':3:log'] == "Output from 3"
+    }
+
+    def "logs at execution time are grouped"() {
+        given:
+        buildFile << """
+            task log { 
+                logger.quiet 'Logged during configuration'
+                doFirst { 
+                    logger.quiet 'First line of text' 
+                    logger.quiet 'Second line of text' 
+                } 
+            }
+        """
+
+        when:
+        succeeds('log')
+
+        then:
+        groupedOutputs[':log'] == "First line of text\nSecond line of text"
+    }
+
+    def "system out and err gets grouped"() {
+        given:
+        buildFile << """
+            task log { 
+                logger.quiet 'Logged during configuration'
+                doFirst { 
+                    System.out.println("Standard out")
+                    System.err.println("Standard err")
+                } 
+            }
+        """
+
+        when:
+        succeeds('log')
+
+        then:
+        groupedOutputs[':log'] == "Standard out\nStandard err"
+    }
+
+    @Issue("gradle/gradle#2038")
+    def "tasks with no actions are not displayed"() {
+        given:
+        buildFile << "task log"
+
+        when:
+        succeeds('log')
+
+        then:
+        groupedOutputs.isEmpty()
+    }
+
+    def "grouped output is displayed for failed tasks"() {
+        given:
+        buildFile << """
+            task log { 
+                logger.quiet 'Logged during configuration'
+                doFirst { 
+                    logger.quiet 'First line of text' 
+                    logger.quiet '' 
+                    logger.quiet '' 
+                    logger.quiet 'Last line of text' 
+                    throw new GradleException('Forced failure')
+                } 
+            }
+        """
+        when:
+        executer.withStackTraceChecksDisabled()
```

Comment:
So I was looking into why this was necessary...

It seems like we're now putting build failures out at stdout, so our scraper treats it like regular output and fails (because the stacktrace _is_ unexpected).  

I'm not sure if that was an intentional change or not